api_app/analyzers_manager/file_analyzers/pe_info.py

```diff
@@ -7,15 +7,104 @@
 # forked repository: https://github.com/mlodic/pefile
 
 import logging
+import os
 from datetime import datetime
 
+import lief
 import pefile
+import pyimpfuzzy
+from PIL import Image
 
 from api_app.analyzers_manager.classes import FileAnalyzer
+from api_app.exceptions import AnalyzerRunException
 
 logger = logging.getLogger(__name__)
 
 
+class WinPEhasher:
+    file_path: str
+
+    def __init__(self, file_path: str, parent=None) -> None:
+        """
+        Init a WinPE instance and load the exe file
+        """
+        self.file_path = file_path
+        self.parent = parent
+
+    def dhashicon(self) -> str:
+        """
+        Return Dhashicon of the exe
+        """
+        try:
+            # this code was implemented from
+            # https://github.com/fr0gger/SuperPeHasher/commit/e9b753bf52d4e48dda2da0b7801075be4037a161#diff-2bb2d2d4d25fef20a893a4e93e96c9b8c0b0c6d5791fc14594cc9dd5cbf40b41
+            #
+            # config
+            hash_size = 8
+            # extract icon
+            icon_path = self.file_path + ".ico"
+            binary = lief.parse(self.file_path)
+            if binary is None:
+                # Invalid PE file
+                self._warn_error("PE_Info.WinPEhasher.dhashicon: invalid_nt_file")
+                return ""
+            # extracting icon and saves in a temp file
+            binres = binary.resources_manager
+            if not binres.has_type(lief.PE.RESOURCE_TYPES.ICON):
+                # no icon resources in file
+                self._warn_error("PE_Info.WinPEhasher.dhashicon: no_icon_res")
+                return ""
```

Comment:
you need to handle these cases with a custom exception.

Reply:
Ok Sir